vars/abapEnvironmentPipelineStageCloneRepositories.groovy

```diff
@@ -8,20 +8,68 @@ import static com.sap.piper.Prerequisites.checkScript
 @Field Set GENERAL_CONFIG_KEYS = []
 @Field STAGE_STEP_KEYS = [
     /** Pulls Software Components / Git repositories into the ABAP Environment instance */
-    'abapEnvironmentPullGitRepo'
+    'abapEnvironmentPullGitRepo',
+    /** Checks out a Branch in the pulled Software Component on the ABAP Environment instance */
+    'abapEnvironmentCheckoutBranch',
+    /** Clones Software Components / Git repositories into the ABAP Environment instance and checks out the respective branches */
+    'abapEnvironmentCloneGitRepo',
+    /** Specifies the strategy that should be peformed on the ABAP Environment instance*/
+    'strategy'
 ]
 @Field Set STEP_CONFIG_KEYS = GENERAL_CONFIG_KEYS.plus(STAGE_STEP_KEYS)
 @Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
 /**
- * This stage clones Git repositories / software components to the ABAP Environment instance
+ * This stage clones Git repositories / software components to the ABAP Environment instance and checks out the master branch
  */
 void call(Map parameters = [:]) {
     def script = checkScript(this, parameters) ?: this
 
     def stageName = parameters.stageName?:env.STAGE_NAME
 
+    Map config = ConfigurationHelper.newInstance(this)
+        .loadStepDefaults()
+        .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+        .mixinStageConfig(script.commonPipelineEnvironment, stageName, STEP_CONFIG_KEYS)
+        .mixin(parameters, PARAMETER_KEYS)
+        .addIfEmpty('strategy', parameters.script.commonPipelineEnvironment.configuration.runStage?.get(stageName))
```

Comment:
I assume this needs to be adapted. Maybe:
```suggestion
        .addIfEmpty('strategy', 'Pull')
```

There is a specific file, which can also be used for Pipeline defaults: resources/com.sap.piper/pipeline/abapStageDefaults.yml

If possible, I would include the default value there.